builtin/credential/ldap/backend.go

```diff
@@ -106,7 +106,7 @@ func (b *backend) Login(req *logical.Request, username string, password string)
 		sresult, err := c.Search(&ldap.SearchRequest{
 			BaseDN: cfg.UserDN,
 			Scope:  2, // subtree
-			Filter: fmt.Sprintf("(%s=%s)", cfg.UserAttr, EscapeLDAPValue(username)),
+			Filter: fmt.Sprintf("(%s=%s)", cfg.UserAttr, ldap.EscapeFilter(username)),
```

Comment:
You're right. Gosh why must these things be so complex. Why did people even use custom grammars for every darn protocol back then?!